src/java_bytecode/java_bytecode_parser.cpp

```diff
@@ -333,19 +333,33 @@ void java_bytecode_parsert::get_class_refs()
     }
   }
 
-  for(const auto &m : parse_tree.parsed_class.fields)
+  for(const auto &f : parse_tree.parsed_class.fields)
   {
-    typet t=java_type_from_string(m.signature);
+    typet t;
+    if(f.has_signature)
+    {
+      t=java_type_from_string(f.signature);
+    }
+    else
+      t=java_type_from_string(f.descriptor);
     get_class_refs_rec(t);
   }
 
   for(const auto &m : parse_tree.parsed_class.methods)
   {
-    typet t=java_type_from_string(m.signature);
+    typet t;
```

Comment:
agreed, but then we will have to split up again when doing this in a try block to fall-back to parsing the descriptor in case we hit a signature we do not support yet